### PR TITLE
docs: improvements to external storage documentation

### PR DESCRIPTION
**Pull request by aoles**
### Pull Request Checklist

- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

File: docs/api-reference/endpoints/directions/routing-options.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ With the request body parameter `options`, advanced routing options can be speci
 The `options` parameter is a JSON object, multiple of the here mentioned parameters can be set. 
 For structure refer to the [examples](#examples).
 
+::: warning
+Some of the options below rely on auxiliary data included on the graph, see [External storages](../../../run-instance/configuration/ors/engine/profiles.md#ext_storages)
```
Comment (koebi) [inline docs/api-reference/endpoints/directions/routing-options.md]:
```suggestion
Some of the options below rely on auxiliary data included in the graph, see [External storages](../../../run-instance/configuration/ors/engine/profiles.md#ext_storages)
```

File: docs/run-instance/configuration/ors/engine/profiles.md [hunk 2]
```diff
@@ -160,8 +160,12 @@ Properties beneath `ors.engine.profiles.*.execution.methods.core`:
 
 ## ext_storages
 
-For each profile it can be defined which external storages for extra info should be included in the graph.
-This makes those information available as `extra_info` in a routing response.
+For each profile it can be defined which auxiliary metadata should be included in the graph.
+This information is made available as `extra_info` in a routing response.
+Additionally, data from `WayCategory` and `Tollways` is being used to filter out certain roads via
+[`avoid_features`](../../../../api-reference/endpoints/directions/routing-options.md#options-avoid-features) api query
+parameter, and `Borders` is necessary for the functionality behind [`avoid_borders`](../../../../api-reference/endpoints/directions/routing-options.md#options-avoid-borders)
```
Comment (koebi) [inline docs/run-instance/configuration/ors/engine/profiles.md]:
```suggestion
Additionally, data from `WayCategory` and `Tollways` is being used to filter out certain roads via
the [`avoid_features`](../../../../api-reference/endpoints/directions/routing-options.md#options-avoid-features) query parameter, and `Borders` is necessary for the functionality behind [`avoid_borders`](../../../../api-reference/endpoints/directions/routing-options.md#options-avoid-borders)
```

File: docs/run-instance/configuration/ors/engine/profiles.md [hunk 3]
```diff
@@ -171,17 +175,18 @@ Properties beneath `ors.engine.profiles.*.ext_storages`:
...
-| TrailDifficulty        | object | Returns the trail difficulty in the route response, Compatible for profile-hiking                                            |                                                   |
+| WayCategory            | object | Returns the way category in the route response, compatible with any profile type                                             |                                                   |
+| WaySurfaceType         | object | Returns the way surface in the route response, compatible with any profile type                                              |                                                   |
+| Tollways               | object | Returns way tolls in the route response, compatible with driving profiles                                                    |                                                   |
+| HillIndex              | object | Returns the ascent/descent in the route response, sompatible with any profile type                                           |                                                   |
```
Comment (koebi) [inline docs/run-instance/configuration/ors/engine/profiles.md]:
```suggestion
| HillIndex              | object | Returns the ascent/descent in the route response, compatible with any profile type                                           |                                                   |
```

**Review by koebi — APPROVED** [review]
Looks good, I commented a few typos/formulation changes.

**Review by MichaelsJP — APPROVED** [review]

**Comment by aoles** [comment]
It turned out tollways storage is not the only one which has not been covered in the docs. It needs to be double checked what is still missing and documented accordingly to finish-up this PR.

**Pull request merged**